### PR TITLE
Convert final-form forms to DDF and get rid of react-ui-components

### PR DESCRIPTION
**Pull request by skateman**
I am trying to get rid of the dependencies to `react-ui-components` from [providers](https://github.com/ManageIQ/manageiq-ui-classic/issues/6716#issuecomment-591974680) so we will be able to get rid of the final-form dependency which is locked to an old version that prevents us from upgrading to DDFv2. In order to do so, I am rewriting the final-form forms to DDFv1.

There are two such forms in this repository, both under the physical servers and accessible from here:
Screenshot from 2020-05-04 18-35-10

The provisioning form loads the options of the second and third dropdown based on a value in the previous dropdowns. There's no support for such thing in DDFv1, so I hacked it around with a dynamic schema and state management. However, there's a bug in DDFv1 that doesn't allow combining dynamic dropdown options with state management and searchability. So in order to have this working, I had to temporarily disable the searchability in these three dropdowns.

Also the second and third dropdown are only displayed if there is a value selected in the dependent dropdown:
Screenshot from 2020-05-04 20-01-46
Screenshot from 2020-05-04 20-01-51
Screenshot from 2020-05-04 20-01-58

These tradeoffs are kinda necessary to move further and they will be addressed after we upgrade to DDFv2.

The firmware upgrade form was much simpler, I only had to make the error handling consistent with the rest of the application like I do in [my analogous PR in the nuage repo](https://github.com/ManageIQ/manageiq-providers-nuage/pull/210). Except of this, the form should be identical to the old one.

There are warnings coming from react on both forms as the dynamic loading and the re-render aren't really friends. This is also fixed in DDFv2, but until then it doesn't break anything, it's just red in the console.

Depends on https://github.com/ManageIQ/manageiq-ui-classic/pull/7005

@miq-bot add_reviewer @himdel
@miq-bot add_label cleanup, react

### CODE REVIEW

**Comment by himdel** [comment]
The code looks good :+1: , but the problem is jest tests.

`import ... from @@ddf` won't work in a plugin jest without loading the jest config from ui-classic.

For the same reason, we can't really remove .babelrc and react&co from dependencies, because local jest relies on them.

(Jest tests were never really a part of the plugin design :(.)

I guess what we can do is not test anything that uses ddf, or to mock ddf for jest.

Also, seems like it now needs a `redux-mock-store` devDependency, I guess it may have been an implicit dependency of react-ui-components  before.

**Comment by himdel** [comment]
Also, the tests still reference removed files:

```
spec/javascript/components/forms/firmware-update.test.jsx
1:import FirmwareUpdate from '../../../../app/javascript/components/forms/firmware-update'

spec/javascript/components/forms/pxe-details.test.jsx
1:import PxeDetails from '../../../../app/javascript/components/forms/pxe-details'

spec/javascript/components/redfish-server-firmware-update-dialog.test.jsx
2:import * as api from '../../../app/javascript/utils/api';

spec/javascript/components/redfish-server-provision-dialog.test.jsx
2:import * as api from '../../../app/javascript/utils/api';
```

**Comment by himdel** [comment]
Hm, after deleting the 2 tests for nonexistent component, what remains are just 2 tests for the dialogs themselves, testing load, and buttons.

Given that, it may make sense to drop jest, babel and all package.json dependencies, devDependencies & scripts.

**Comment by himdel** [comment]
`Rakefile` still calls it

**Comment by agrare** [comment]
@skateman after merging this I'm getting the following error:

```
error An unexpected error occurred: "/home/grare/adam/.gems/2.7.0/bundler/gems/manageiq-providers-redfish-ea8916bcc8c4/package.json: Unexpected token } in JSON at position 600".
info If you think this is a bug, please open a bug report with the information provided in "/home/grare/adam/.gems/2.7.0/bundler/gems/manageiq-providers-redfish-ea8916bcc8c4/yarn-error.log".
```

Think you have a trailing `,` in the `package.json`

File: package.json [hunk 1]
```diff
@@ -13,47 +13,10 @@
...
-    "enzyme-to-json": "^3.3.4",
-    "identity-obj-proxy": "^3.0.0",
-    "jest": "^23.4.2"
-  },
+  "dependencies": {},
   "engines": {
     "node": ">= 6.9.1",
     "npm": ">= 3.10.3",
     "yarn": ">= 0.20.1"
   },
```
Comment (himdel) [inline package.json:21]: Sorry, missed the  trailing coma :(

```
error An unexpected error occurred: "/home/himdel/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/bundler/gems/manageiq-providers-redfish-ea8916bcc8c4/package.json: Unexpected token } in JSON at position 600".
```

=> https://github.com/ManageIQ/manageiq-providers-redfish/pull/110

**Pull request merged**